framework/web/UrlRule.php

```diff
@@ -158,7 +158,8 @@ public function init()
             $this->name = $this->pattern;
         }
 
-        $this->pattern = trim($this->pattern, '/');
+        // don't trim left slashes if more then one slash at the beginning of pattern
+        $this->pattern = preg_replace('/(^(\/)(?!\/)|(\/$))/', '', $this->pattern);
```

Comment:
Also it will ignore `////test/` while it's not protocol relative URL but definitely a bogus input or something.